drivers/pwm/pwm_mcux_ctimer.c

```diff
@@ -0,0 +1,180 @@
+/*
+ * (c) Meta Platforms, Inc. and affiliates.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+#define DT_DRV_COMPAT nxp_ctimer_pwm
+
+#include <errno.h>
+#include <fsl_ctimer.h>
+#include <fsl_clock.h>
+#include <zephyr/drivers/pwm.h>
+#ifdef CONFIG_PINCTRL
+#include <zephyr/drivers/pinctrl.h>
+#endif
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(pwm_mcux_ctimer, CONFIG_PWM_LOG_LEVEL);
+
+#define CHANNEL_COUNT kCTIMER_Match_3 + 1
+
+#define SECURE_NONSECURE_MSK            (0xF0000000U)
+#define SECURE_NONSECURE_SHIFT          (28U)
+#define PERIPHERAL_BASE_SECURE_ALIAS    (0x05U << SECURE_NONSECURE_SHIFT)
+#define PERIPHERAL_BASE_NONSECURE_ALIAS (0x04U << SECURE_NONSECURE_SHIFT)
+#define PERIPHERAL_BASE		(DT_RANGES_PARENT_BUS_ADDRESS_BY_IDX(DT_NODELABEL(peripheral), 0))
+
+#if (PERIPHERAL_BASE & SECURE_NONSECURE_MSK) == PERIPHERAL_BASE_SECURE_ALIAS
+#define SECURITY_ALIAS PERIPHERAL_BASE_SECURE_ALIAS
+#elif (PERIPHERAL_BASE & SECURE_NONSECURE_MSK) == PERIPHERAL_BASE_NONSECURE_ALIAS
+#define SECURITY_ALIAS PERIPHERAL_BASE_NONSECURE_ALIAS
```

Comment:
why do we have these security aliases defined here? shouldn't the peripheral address come from DT?

Reply:
yeah I can make mask out all but top nibble of peripheral base. peripheral base seems to be where security alias is defined in dt